client/blocks/taxonomy-manager/list-item.jsx

```diff
@@ -42,29 +42,17 @@ class TaxonomyManagerListItem extends Component {
 	};
 
 	state = {
-		popoverMenuOpen: false,
 		showDeleteDialog: false,
 	};
 
-	togglePopoverMenu = event => {
-		if ( event && event.stopPropagation ) {
-			event.stopPropagation();
-		}
-		this.setState( {
-			popoverMenuOpen: ! this.state.popoverMenuOpen
-		} );
-	};
-
-	editItem = () => {
-		this.setState( {
-			popoverMenuOpen: false
-		} );
+	editItem = closeMenu => {
+		closeMenu();
```

Comment:
I think your calling `closeMenu` here is an example of a bug discovered in #7616 where setting `onClick` of a `<PopoverMenuItem />` doesn't dismiss the menu properly. Do you think it would make sense to cherry-pick my proposed fix in 2d2295fb79728d0829f0ba4a2ae2dded25e254aa to your branch?